hilti/toolchain/src/ast/node.cc

```diff
@@ -115,8 +118,10 @@ void Node::replaceChildren(ASTContext* ctx, const Nodes& children) {
 
 void Node::clearChildren() {
     for ( auto& c : _children ) {
-        if ( c )
-            c->_clearParent();
+        if ( c ) {
+            c->_parent = nullptr;
```

Comment:
Do we need this update at all if we destroy all children after the loop anyway?

```suggestion
```

Reply:
It's not guaranteed that we'll destroy after the loop: somebody might re-pin it in the meantime, e.g., by re-adding the node to the AST elsewhere. And having a null parent is the indicator that the node isn't part of the AST yet and doesn't need to be deep-copied when it's added.